contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,249 @@
+This will be achieved by federated controller creating and monitoring individual services in specific clusters (not a federated service) against each pod, with the same name as that pod&#39;s ordinal identity.
+It will further need to be ensured that the pods be assigned the same hostname as the pod name (ordinal identity) and service name to keep it consistent with the in-cluster statefulsets topology.
+
+There is one interesting catch about the service creation, here.
+The service needs to exist locally in the cluster (1 service per replica pod across clusters), which is unlike a federated service.
+If we use the existing federated service mechanism, to create this service, and the service with a particular name (for example web-0 in example above) is created in federation, it will propagate the same to all federated clusters.
+This is unnecessary for the solution mentioned in this section (or for further reading in the following section), and would be extreme waste of resources.
+Some simple proposals to tackle this problem are mentioned at the end of this document under section &quot; **Handling additional LB Services**&quot;.
+
+## Design Alternative 2
+
+In this approach, the federated statefulset controller will behave quite similar to federated replicaset or the federated deployment controller.
+The federated controller would create and monitor individual statefulsets (rather then pods directly) partitioning and distributing the total stateful replicas across the federated clusters.
+
+As a proposal in this design we suggest the possibility of the pods having multiple identities.
+This facilitates application pods being able to participate in different quora, and leaves a choice with application replicas to join either the local or the global quora, as need be.
+As of now we can envision the need of two possible identities for each pod (relevant to use case 2), enabled by this approach.
+The ordinal number assigned to the pod will be calculated and assigned in exactly the same way, it is assigned in the k8s cluster as described [here](https://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/#pod-identity) and will be unique.
+It however will be possible for each pod to take multiply dns identities.
+This is detailed in the &#39;instance identity and discovery&#39; section below.
+
+For this phase of implementation, it is proposed that the statefulset need not guarantee the order of creation of the pod instances across federation (_we believe more discussion will follow this statement_).
+
+### Replica distribution (across federated clusters)
+
+The proposed default behaviour of the federation controller, when a statefulset creation request is sent to federation API server, would be to partition the statefulset replicas and create a statefulset into each of the clusters with the reduced replica number (after partitioning), quite similar to the behaviour of the replicaset or daemonset controllers of the k8s federation.
```

Comment:
The point is that the federation user needs to see a consolidated view of the statefulset. Also, the current k8s federation design philosophy is that the federation ideally appears as another k8s cluster, where some users might not even know that they are talking to federation and not a normal k8s cluster.